### ISSUE TITLE
Question about multiplexing socket.io

### ISSUE THREAD

**Issue by boschkundendienst** [hakimel/reveal.js#2449]
Hi, I got multiplexing working and I am currently doing some test runs leaving a master running 24/7 over mutliple days showing a presentation that loops and automatically turns over slides while multiple clients show the presentation.

Over the last couple of days I had some strange behavious that could either be fixed by reloading the presentation in the browser who is the master or restarting reveal.js which I have implemented as a daemon on my linux host.

With "strange behaviour" I mean, that either the master or the client view quickly turns slides arround or freezes.

I have installed reveal.js using npm and run it as a service from a virtual node environment. Apache proxies requests to reveal.js and socket.io with a configuration like this:

```
<VirtualHost *:80>
DocumentRoot /opt/revealjs/reveal.js/
Options -Indexes

ProxyPass /reveal.js http://127.0.0.1:8000/ flushpackets=on
ProxyPassReverse /reveal.js http://127.0.0.1:8000/
ProxyPass /socket.io http://127.0.0.1:1948/socket.io flushpackets=on
ProxyPassReverse /socket.io  http://127.0.0.1:1948/socket.io
</VirtualHost>
```
I already tried without the `ProxyPassReverse` and without `flushpackets=on`

I found [this interesting question](https://stackoverflow.com/a/9628692/728852) about socket.io at stackoverflow and would kindly ask if the implementation in reveal.js is straight forward and is using the best practizes they give there?

[https://stackoverflow.com/a/9628692/728852](https://stackoverflow.com/a/9628692/728852)

Of course it could be a total different thing but I suspect the socket system or the implementation as a possible cause.

Have you been running multiplexing over multiple days without problems?

P.S.
As many others already told, the description, how to setup multiplexing is a little bit hard to follow and assumes a lot of knowledge about node. When I have my solution ready (reveal.js on CentOS with socket.io, both running as daemon) I will make it public available for others.

Thanks in advance and keep on going with that interesting piece of software.

Regards

Peter

**Comment by cerisara** [comment]
Hi,

I'm very sorry, I don't know any answer to your question, but looking at your apache config, I have a related one: when I launch the multiplex plugin / socket.io server, the master presentation raises cross-domain errors CORS... How have you avoided that ?
Everything works fine with the https://reveal-js-multiplex-ccjbegmaii.now.sh domain, but with mine, I just got CORS errors. So I guess it's either an apache config or the plugin server...

but my apache is as simple as yours... Any hint please ?
Thanks !
